tests/acceptance/30_custom_promise_types/09_unless.cf

```diff
@@ -0,0 +1,129 @@
+######################################################
+#
+#  Test that custom promises can use unless attribute
+#
+#####################################################
+body common control
+{
+    inputs => { "../default.cf.sub" };
+    bundlesequence  => { default("$(this.promise_filename)") };
+    version => "1.0";
+}
+
+#######################################################
+
+bundle agent init
+{
+  files:
+    "$(G.testfile)-0"
+      delete => init_delete;
+    "$(G.testfile)-1"
+      delete => init_delete;
+    "$(G.testfile)-2"
+      delete => init_delete;
+    "$(G.testfile)-3"
+      delete => init_delete;
+    "$(G.testfile)-4"
+      delete => init_delete;
+    "$(G.testfile)-5"
+      delete => init_delete;
+    "$(G.testfile)-6"
+      delete => init_delete;
+    "$(G.testfile)-7"
+      delete => init_delete;
+    "$(G.testfile)-8"
+      delete => init_delete;
+    "$(G.testfile)-9"
+      delete => init_delete;
+    "$(G.testfile)-10"
+      delete => init_delete;
```

Comment:
This could use an slist (with just the indices).

Reply:
This is the same as the if test.